packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -1340,9 +1642,24 @@ export class Combobox
           // eslint-disable-next-line react/jsx-sort-props -- ref should be last so node attrs/props are in sync (see https://github.com/Esri/calcite-design-system/pull/6530)
           ref={this.setReferenceEl}
         >
-          <div class="grid-input">
+          <div
+            class={{
+              "grid-input": true,
+              [CSS.displayModeFitToLine]: fitToLineDisplayMode,
+              [CSS.displayModeSingle]: singleDisplayMode,
+            }}
+            ref={this.setChipContainerEl}
+          >
             {this.renderIconStart()}
-            {!single && this.renderChips()}
+            {!singleSelectionMode && !singleDisplayMode && this.renderChips()}
```

Comment:
Can you look at simplifying this? IMO, it's kind of hard to follow because some variables are similarly named and there's some duplication going on.  

One option would be to render multiple parts that share the same condition:

```tsx
{!singleSelectionMode &&
  !showAllDisplayMode && [
    this.renderSelectedIndicatorChip(),
    this.renderSelectedIndicatorChipCompact(),
    this.renderAllSelectedIndicatorChip(),
    this.renderAllSelectedIndicatorChipCompact(),
  ]}
```